### PR TITLE
Add MultiSelect Design-spec for new `onTagsAdded` API

### PR DESCRIPTION
**Pull request by hepiyellow**

### CODE REVIEW

File: src/MultiSelect/README.DESIGN-SPEC.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+|1|`{ id: 'Alabama', value: 'Alabama' }` |`{ id: 'Alabama' }` |
+|2|`{ id: 'Arkansas', value: <span>Arkansas</span> }` | `{ id: 'Arkansas' }` |
+|3|`{ id: 'California', value: 'California', code: 'CA' }` | `{ id: 'California', code: 'CA'}`|
+
+> If the consumer wants, they can put a `tag` property on the option (which is opaque to MultiSelect), and use it upon selection to add to the tags.
+
+### Paste logic
+
+- Paste will automatically submit (call `onTagsAdded`).
+- Paste will automatically clear the input.
```
Comment (yardnsm) [inline src/MultiSelect/README.DESIGN-SPEC.md]: Do we really want this? Users may paste a partial value, and then want to edit it before submitting.

File: src/MultiSelect/README.DESIGN-SPEC.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+| markFirstMatchingOption | boolean | false |  | When true then when typing, the first matching option will be "marked", so that pressing "Enter" will select it.|
+| onSelect | function | | |`onSelect` will be called ONLY when a user clicks on an option from the options list. It receives an `Array<object>` of newly selected options (usually one). Each object is the original option, excluding the `value` property. (`value` property should be used for rendering the option only)|
+
+### markFirstMatchingOption (new prop)
+
+For example, when there is an `Alabama` option, and the input contains `Alab`, the options is showing the `Alabama` option.
+
+- `markFirstMatchingOption === false` (default) : No option will be marked, "Enter" submits (calls `onTagsAdded`) the input value. (The user can navigate to the options)
+- `markFirstMatchingOption === true` : The first matching option is marked (like hovered), "Enter" selects it.
```
Comment (yardnsm) [inline src/MultiSelect/README.DESIGN-SPEC.md]: So when `markFirstMatchingOption` is `true`, the user can't add new tags which are a subset of existing options, BUT can sill add new tags that are not a subset?

(I also read the Callback summary below)

File: src/MultiSelect/README.DESIGN-SPEC.md (hunk 1 above)
Comment (hepiyellow) [reply src/MultiSelect/README.DESIGN-SPEC.md]: Ben says this is a rare case.
The consumer can implement this by ignoring the submit (according to some business logic) and set the `value` according to the submitted value.

Comment (hepiyellow) [reply src/MultiSelect/README.DESIGN-SPEC.md]: Plus, we can change this later and it won't be a breaking change.

File: src/MultiSelect/README.DESIGN-SPEC.md (hunk 2 above)
Comment (hepiyellow) [reply src/MultiSelect/README.DESIGN-SPEC.md]: Yes. 
It is actually also the behaviour of Gmail Compose "To" input box.

**Pull request merged**